packages/react/src/useActionForm.ts

```diff
@@ -0,0 +1,263 @@
+ *
+ * `useActionForm` doesn't run the action when called. To actually submit the form, call the `submit` function on the returned `Form` object.
+ *
+ * @see https://react-hook-form.com/
+ *
+ * @param action an action function from a model manager in your application's client, like `api.user.create`
+ * @param options action options, like selecting the fields in the result
+ */
+export const useActionForm = <
+  GivenOptions extends OptionsType, // currently necessary for Options to be a narrow type (e.g., `true` instead of `boolean`)
+  SchemaT,
+  ActionFunc extends ActionFunction<GivenOptions, any, any, SchemaT, any>,
+  FormContext = any
+>(
+  action: ActionFunc,
+  options?: UseFormProps<ActionFunc["variablesType"], FormContext> & {
+    /**
+     * The record identifier to run this action on, if it already exists.
+     * Should be undefined for create actions, or a record ID (or finder) for update / etc actions
+     **/
+    findBy?: RecordIdentifier;
+    /**
+     * The record to run this action on if it has already been fetched.
+     * Should be undefined for create actions, or a record for update / etc actions
+     * If no record is available, use `findBy` instead.
+     */
+    record?: ActionFunc["hasReturnType"] extends true
+      ? any
+      : Partial<
+          GadgetRecord<
+            Select<
+              Exclude<ActionFunc["schemaType"], null | undefined>,
+              DefaultSelection<ActionFunc["selectionType"], ActionFunc["optionsType"], ActionFunc["defaultSelection"]>
+            >
+          >
+        >;
+    /**
+     * Which fields to select on from the record when retrieving it from the backend.
+     */
+    select?: ActionFunc["optionsType"]["select"];
+    /**
+     * Which fields to send from the form's values when sending it from the backend.
+     */
+    send?: string[];
+    /**
+     * Called when the form submits
+     */
+    onSubmit?: () => void;
+    /**
+     * Called when the action completes successfully on the backend
+     */
+    onSuccess?: (actionResult: any) => void;
+    /**
+     * Called when the form submission errors before sending, during the API call, or if the API call returns an error.
+     */
+    onError?: (error: Error | FieldErrors<ActionFunc["variablesType"]>) => void;
+  }
+): UseActionFormResult<GivenOptions, SchemaT, ActionFunc, FormContext> => {
+  const api = useApi();
+  const findExistingRecord = !!options?.findBy;
+  const hasSetInitialValues = useRef<boolean>(!findExistingRecord);
+
+  // find the existing record if there is one
+  const modelManager = (api as any)[action.modelApiIdentifier] as AnyModelManager;
+  const [findResult] = useFindExistingRecord(modelManager, options?.findBy || "1", {
```

Comment:
Is "1" the correct default we want for this?

Reply:
I think this is a hack since we always need to pass a value in here; if there is no `findBy` then the hook is always paused

Reply:
A hack indeed, maybe we should just make the value optional and that'd be clearer?

Reply:
hm yea. the nice thing about this is it works if a user goes from not passing in a string to passing one in. won't work for `{ [key: string]: any }` findBys though